tests/slurm_command_gen_strategy/test_nemo_launcher.py

```diff
@@ -135,21 +135,13 @@ def test_gpus_per_node_value(self, nemo_cmd_gen: NeMoLauncherSlurmCommandGenStra
 
         assert "cluster.gpus_per_node=null" in cmd
 
-    def test_argument_with_tilde_value(self, nemo_cmd_gen: NeMoLauncherSlurmCommandGenStrategy, nemo_test_run: TestRun):
-        tdef: NeMoLauncherTestDefinition = cast(NeMoLauncherTestDefinition, nemo_test_run.test.test_definition)
-        tdef.cmd_args.training.model.micro_batch_size = "~"  # type: ignore
-
-        cmd = nemo_cmd_gen.gen_exec_command(nemo_test_run)
-        assert "~training.model.micro_batch_size=null" in cmd
-
     def test_data_impl_mock_skips_checks(
         self, nemo_cmd_gen: NeMoLauncherSlurmCommandGenStrategy, nemo_test_run: TestRun
     ):
-        tdef: NeMoLauncherTestDefinition = cast(NeMoLauncherTestDefinition, nemo_test_run.test.test_definition)
-        tdef.cmd_args.data_dir = "DATA_DIR"
-
-        cmd = nemo_cmd_gen.gen_exec_command(nemo_test_run)
-        assert "data_dir" in cmd
+        with patch.object(Test, "extra_cmd_args", new_callable=PropertyMock) as mock_extra_cmd_args:
```

Comment:
Can we assign a value to `nemo_test_run.test....` directly instead of patching it? Mocks are hiding real behaviour.

Reply:
I tried to assign a value to test_definition.extra_cmd_args_str, but it failed, indicating that I was not allowed to do so. I assumed it was not allowed because of the pydantic implementation.

Reply:
Right, but you should be able to assign values to `tdef.extra_cmd_args`, it is a simple dictionary.

Reply:
Updated, thanks.